hexrd/ui/calibration/auto/instrument_calibrator.py

```diff
@@ -19,43 +41,112 @@ def instr(self):
     def calibrators(self):
         return self._calibrators
 
+    @property
+    def flags(self):
+        # additional params come next
+        flags = [self.instr.calibration_flags, ]
+        for calib_class in self.calibrators:
+            flags.append(calib_class.flags[calib_class.npi:])
+        return np.hstack(flags)
+
+    @property
+    def reduced_params(self):
+        return self.full_params[self.flags]
+
     # =========================================================================
     # METHODS
     # =========================================================================
 
+    def _reduced_params_flag(self, cidx):
+        assert cidx >= 0 and cidx < len(self.calibrators), \
+            "index must be in %s" % str(np.arange(len(self.calibrators)))
+
+        calib_class = self.calibrators[cidx]
+
+        # instrument params come first
+        npi = calib_class.npi
+
+        # additional params come next
+        cparams_flags = [calib_class.flags[:npi], ]
+        for i, calib_class in enumerate(self.calibrators):
+            if i == cidx:
+                cparams_flags.append(calib_class.flags[npi:])
+            else:
+                cparams_flags.append(np.zeros(calib_class.npe, dtype=bool))
+        return np.hstack(cparams_flags)
+
+    def extract_points(self, fit_tth_tol, int_cutoff=1e-4):
+        # !!! list in the same order as dict looping
+        master_data_dict_list = []
+        for calib_class in self.calibrators:
+            master_data_dict_list.append(
+                calib_class._extract_powder_lines(
+                    fit_tth_tol=fit_tth_tol, int_cutoff=int_cutoff
+                )
+            )
+        return master_data_dict_list
+
+    def residual(self, x0, master_data_dict_list):
```

Comment:
Looks like the argument `x0` is not being used. Is that okay?

Reply:
Nope!   Fixed.